frontend/app/assets/javascripts/spree/frontend/views/spree/shared/variant_select.js

```diff
@@ -0,0 +1,87 @@
+let getQueryString = window.location.search
+let urlParams = new URLSearchParams(getQueryString)
+let variantIdFromUrl = urlParams.get('variant')
+
+this.initializeQueryParamsCheck = function () {
+  if (urlParams.has('variant')) {
+    this.variants.forEach(varifyVariantID)
+  }
+}
+
+function varifyVariantID() {
+  for (const variant of this.variants) {
+    if (parseInt(variant.id) === parseInt(variantIdFromUrl)) {
+      this.urlQueryMatchFound = true
+    }
+  }
+}
+
+this.setSelectedVariantFromUrl = function () {
+  this.selectedOptions = []
+  this.getVariantOptionsById(variantIdFromUrl)
+  this.selectedOptions.sort((a, b) => (a.dataset.optionTypeIndex > b.dataset.optionTypeIndex) ? 1 : -1)
+  this.clickListOptions(this.selectedOptions)
+  this.updateStructuredData()
+}
+
+this.getVariantOptionsById = function(variantIdFromUrl) {
+  for (const v of this.variants) {
+    if (parseInt(v.id) === parseInt(variantIdFromUrl)) {
+      this.sortOptionValues(v.option_values)
+    }
+  }
+}
+
+this.sortOptionValues = function(optVals) {
+  optVals.forEach(buidArray)
+}
+
+function buidArray(item) {
+  const container = document.querySelector('ul#product-variants')
+  const target = container.querySelectorAll('.product-variants-variant-values-radio')
+
+  for (const inputTag of target) {
+    if (parseInt(inputTag.value) === item.id && inputTag.dataset.presentation === item.presentation) {
+      this.selectedOptions.push(inputTag)
+    }
+  }
+}
+
+this.clickListOptions = function(list) {
+  list.forEach(selectOpts)
+
+  function selectOpts(item, index) {
+    item.click()
+    var $t = $(item)
```

Comment:
Could You change `$t` to a more meaningful name?

Reply:
Fixed.